src/Streamly/Data/Fold.hs

```diff
@@ -116,143 +173,82 @@ module Streamly.Data.Fold
     , and
     , or
 
-    -- * Output Transformations
-    -- | Unlike stream producer types (e.g. @SerialT m a@) which have only
-    -- output side, folds have an input side as well as an output side.  In the
-    -- type @Fold m a b@, the input type is @a@ and the output type is @b@.
-    -- Transformations can be applied either on the input side or on the output
-    -- side. The 'Functor' instance of a fold maps on the output of the fold:
+    -- * Combinators
+    -- | Combinators are modifiers of folds.  In the type @Fold m a b@, @a@ is
+    -- the input type and @b@ is the output type.  Transformations can be
+    -- applied either on the input side or on the output side.  Therefore,
+    -- combinators are of one of the following general shapes:
+    --
+    -- * @... -> Fold m a b -> Fold m c b@ (input transformation)
+    -- * @... -> Fold m a b -> Fold m a c@ (output transformation)
+    --
+    -- Output transformations are also known as covariant transformations, and
+    -- input transformations are also known as contravariant transformations.
+    -- The input side transformations are more interesting for folds.  Most of
+    -- the following sections describe the input transformation operations on a
+    -- fold.  The names and signatures of the operations are consistent with
+    -- corresponding operations in "Streamly.Prelude". When an operation makes
+    -- sense on both input and output side we use the prefix @l@ (for left) for
+    -- input side operations and the prefix @r@ (for right) for output side
+    -- operations.
+
+    -- ** Mapping on output
+    -- | The 'Functor' instance of a fold maps on the output of the fold:
     --
     -- >>> Stream.fold (fmap show Fold.sum) (Stream.enumerateFromTo 1 100)
     -- "5050"
     --
-    -- Note: Output transformations are also known as covariant
-    -- transformations.
-    , sequence
     , rmapM
-    , mapM
-
-    {-
-    -- * Input Transformations
-    -- The input side transformations are more interesting for folds.  The
-    -- following sections describe the input transformation operations on a
-    -- fold.  The names and signatures of the operations are consistent with
-    -- corresponding operations in "Streamly.Prelude".
-    --
-    -- Note: Input transformations are also known as contravariant
-    -- transformations.
-
-    -- ** Mapping
-    --, transform
-    -- , lmap
-    --, lsequence
-    -- , lmapM
-
-    -- -- ** Filtering
-    -- , filter
-    -- , filterM
-    -- , ldeleteBy
-    -- , luniq
-
-    -- ** Mapping Filters
-    , lmapMaybe
-    , lmapMaybeM
-
-    -- ** Scanning Filters
-    , lfindIndices
-    , lelemIndices
 
-    -- ** Insertion
-    -- | Insertion adds more elements to the stream.
+    -- ** Mapping on Input
+    , lmap
+    , lmapM
 
-    , linsertBy
-    , lintersperseM
+    -- ** Filtering
+    , filter
+    , filterM
 
-    -- ** Reordering
-    , lreverse
-    -}
+    -- -- ** Mapping Filters
```

Comment:
Extra `--`?

Reply:
That is intentional to comment out the haddock markup.